prusti-viper/src/encoder/encoder.rs

```diff
@@ -1680,6 +1687,29 @@ impl<'v, 'tcx> Encoder<'v, 'tcx> {
         };
         result
     }
+
+    pub fn add_discriminant_info(&self, enum_id: String, discr_id: String, proc_def_id: ProcedureDefId) {
+        let mut map = self.discriminants_info.borrow_mut();
+        let mut previous_entries = map.get_mut(&(proc_def_id.clone(), enum_id.clone()));
+        match previous_entries {
+            None => {map.insert((proc_def_id, enum_id), vec![discr_id]);},
+            Some(v) => {
+                v.push(discr_id);
+            }
```

Comment:
```suggestion
            None => map.insert((proc_def_id, enum_id), vec![discr_id]),
            Some(v) => v.push(discr_id),
```

Reply:
The second pair of curly brackets are totally unnecessary, I agree, but `map.insert()` returns an Option, so there your suggestion does not work because "match arms have incompatible types".